### PR TITLE
Report structured repository revision snapshots

### PR DESCRIPTION
**Pull request by szgupta**
## Summary
- capture the resolved HEAD for each structured environment repository inside the existing clone/reuse command
- assemble one immutable execution snapshot with bounded unresolved-repository diagnostics
- deliver the typed payload on a detached background task with 3-second per-attempt timeouts and bounded transient retries
- emit an explicit empty snapshot for executions with no structured repositories

## Non-blocking design
Each repository writes to a unique `.warp/repository-revisions/<snapshot_uuid>/<index>.head` path through a temporary file plus atomic rename. Capture commands end in a best-effort fallback, so revision lookup or file-write failures cannot alter clone success. Snapshot parsing omits missing/invalid SHA-1 or SHA-256 IDs and cleans up result directories best effort. HTTP serialization, timeout handling, and retries happen only after `Background::spawn(...).detach()`; no network future is awaited by environment preparation.

## Wire payload
`POST /api/v1/agent/runs/{run_id}/repository-revisions`

Top level: `snapshot_uuid`, `captured_at`, `unresolved_repository_count`, `repositories`.

Repository: `code_forge`, `owner`, `repo`, `checkout_path`, optional `checkout_ref`, `head_sha`.

## Validation
Passed:
- `cargo fmt --all --manifest-path /workspace/warp/Cargo.toml -- --check`
- `git -C /workspace/warp --no-pager diff --check`
- `cargo metadata --manifest-path /workspace/warp/Cargo.toml --no-deps --format-version 1`
- real-Git shell verification of atomic HEAD capture and non-fatal capture failure

Sandbox memory limitation (all reached the monolithic `warp` crate, emitted no Rust diagnostic, and were killed by SIGKILL):
- `cargo check --manifest-path /workspace/warp/Cargo.toml -p warp --lib`
- `CARGO_BUILD_JOBS=1 cargo test --manifest-path /workspace/warp/Cargo.toml -p warp --lib git_object_id_validation_accepts_sha1_and_sha256_only --no-run`
- `CARGO_BUILD_JOBS=1 cargo check --manifest-path /workspace/warp/Cargo.toml -p warp --lib --no-default-features`
- `CARGO_BUILD_JOBS=1 CARGO_PROFILE_DEV_DEBUG=0 CARGO_PROFILE_DEV_INCREMENTAL=false cargo check --manifest-path /workspace/warp/Cargo.toml -p warp --lib`

_Plans_:
- _[Emit repository revision snapshots for Warp agent runs](https://staging.warp.dev/drive/notebook/WNbgAMWYk8Fd2fTkB5EjVB)_

Co-Authored-By: Warp <agent@warp.dev>

### CODE REVIEW

**Comment by szgupta** [comment]
This PR was generated with [Warp](https://warp.dev/oz).

_Comment `@warp-agent` on this PR to send it follow-up work._


<div><a href="https://oz.staging.warp.dev/runs/01a01c5f-9925-7215-a2db-f1b436517b14"><picture><source media="(prefers-color-scheme: dark)" srcset="https://oz.warp.dev/images/github-buttons/view-run-dark.png">View run</picture></a>&nbsp;<a href="https://staging.warp.dev/conversation/a3390b46-6295-4a51-9dc5-1ddb9f87bf16"><picture><source media="(prefers-color-scheme: dark)" srcset="https://oz.warp.dev/images/github-buttons/view-conversation-dark.png">View conversation</picture></a></div>
